### ISSUE TITLE
Genambpos

### ISSUE THREAD

**Issue by sariths** [gmischler/PyRad#7]
Hi @gmischler , 

I was trying to understand a few things about ambient calculations when I came across [Genambpos by John Mardaljevic](http://radiance-online.org/cgi-bin/viewcvs.cgi/ray/src/util/genambpos.pl?revision=2.7&view=markup). As there wasn't a cross-platform version available, I decided to rewrite his script in Python. My version (still WIP) is here: https://github.com/sariths/PyRad/blob/master/scripts/genambpos.py. My script trips when it invokes rcalc here: https://github.com/sariths/PyRad/blob/master/scripts/genambpos.py#L149-L164. The error is 

```
-f
   ^ illegal variable name
```

 Any idea why this might be? I tried playing around with the quotes but that did not make any difference.

I don't have much of any idea of what the options in the script mean as the original version wasn't documented comprehensively enough for someone like me to understand. I was hoping to tinker with the script and then rename variables like `minwt`, `direct`,`position` etc after understanding what they imply.

Thanks,
Sarith

**Comment by gmischler** [comment]
@sariths, the first thing jumping at me: Your `self.radius` is another `-e` option with its own argument. This should not be a single string, but a list to splice into the complete argument list:

`self.radius = ['-e', 'psiz:%s' % args.radius] if args.radius else []`

You should generally treat the `-e` option and its respective arguments as seperate items:

``` python
    rcalcCmd = ['rcalc',
        '-e', 'LV:{0};MW:{1};SF:{2}'.format(self.level, self.minwt, self.scalingFactor),
        '-f', 'rambpos.cal',
        '-e', 'cond=acond', ]
        + self.radius
        + ['-o', '%s' % posGradFormat, ]
```

Not sure if that's the actual cause of your error message (not tested). But it's something you should fix before we can investigate further.
Btw: If you leave a space after each comma, the code is much easier to read.

**Comment by sariths** [comment]
@gmischler, The argument thing got rid of a lot of errors. The script now kind of works (its still incomplete) but there are some zero division errors and cones with duplicate centers now. I will try and figure out and check with you if I can't move forward. Thanks a lot for the quick and precise response.

Sarith

**Comment by sariths** [comment]
Hi @gmischler , just  a quick update. I got the script working. The zero division errors that I previously mentioned were due to some misplaced strings. The updated script is on [my fork](https://github.com/sariths/PyRad/blob/master/scripts/genambpos.py). I haven't done the dry run, verbose features etc. yet.  It is, however, cross-platform and works on Unix/Windows unlike the original perl script (I guess the .pl script could be made cross-platform if something other than `sed` is used for extracting `-aa` value). I will include this with the other scripts and send a combined pull request eventually. Thanks again for the help.
![5-20-2016 12-37-33 am](https://cloud.githubusercontent.com/assets/10361848/15417494/823a3174-1e24-11e6-976b-7a45e8a1f184.png)
![5-20-2016 12-32-30 am](https://cloud.githubusercontent.com/assets/10361848/15417496/84bcfba2-1e24-11e6-9b32-93b3673222eb.png)

Sarith
